In [165]:
import pandas as pd
import numpy as np

# 2005-2006 Data

In [166]:
# 2005-2006
path = '/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/2005-2006'

import os
os.chdir(path)

## PHQ-9 Survey - part of Questionnaire

In [167]:
DPQ_2005_2006 = pd.read_csv('./Questionnaire/DPQ_D.csv')

In [168]:
DPQ_2005_2006.head()

SEQN        DPQ010        DPQ020        DPQ030        DPQ040  \
0  31130.0           NaN           NaN           NaN           NaN   
1  31131.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  31132.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  31134.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
4  31139.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   

         DPQ050        DPQ060        DPQ070        DPQ080        DPQ090  \
0           NaN           NaN           NaN           NaN           NaN   
1  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
4  3.000000e+00  1.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79   

         DPQ100  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4  5.397605e-79

In [169]:
# drop final follow-up question
DPQ_2005_2006.drop(['DPQ100'], axis=1, inplace=True)

In [170]:
# values less than 1 are zero
def set_zero(df):
    df[df<1] = 0
    
set_zero(DPQ_2005_2006)

In [171]:
# need to drop rows where not all col are complete
DPQ_2005_2006.dropna(axis=0, how='any', thresh=None, subset=None, inplace=True)

In [172]:
# take a sum across rows and save as new col
DPQ_2005_2006['sum'] = DPQ_2005_2006.iloc[:,1:].sum(axis=1)

In [173]:
# create new target label 'depressed' if sum is greater than 10 --> 1, otherwise 0
DPQ_2005_2006.loc[DPQ_2005_2006['sum'] > 10, 'depressed'] = 1
DPQ_2005_2006.loc[DPQ_2005_2006['sum'] <= 10, 'depressed'] = 0

In [174]:
DPQ_2005_2006.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
1  31131.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2  31132.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3  31134.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
4  31139.0     0.0     0.0     0.0     0.0     3.0     1.0     0.0     0.0   
5  31143.0     0.0     1.0     1.0     1.0     1.0     1.0     1.0     0.0   

   DPQ090  sum  depressed  
1     0.0  0.0        0.0  
2     0.0  0.0        0.0  
3     0.0  0.0        0.0  
4     0.0  4.0        0.0  
5     0.0  6.0        0.0

In [175]:
DPQ_2005_2006.shape

(4836, 12)

In [176]:
DPQ_D = DPQ_2005_2006[['SEQN', 'depressed']]

In [177]:
DPQ_D.head()

SEQN  depressed
1  31131.0        0.0
2  31132.0        0.0
3  31134.0        0.0
4  31139.0        0.0
5  31143.0        0.0

In [178]:
len(DPQ_2005_2006[DPQ_2005_2006['depressed']==1])  # about 8% depressed = true

268

## Demographic Data

### Demographic Variables and Sample Weights (DEMO_D)

In [179]:
# start with demographic data
DEMO_2005_2006 = pd.read_csv('./Demographics/DEMO_D.csv')

In [180]:
DEMO_2005_2006.head(10)

SEQN  SDDSRVYR  RIDSTATR  RIDEXMON  RIAGENDR      RIDAGEYR  RIDAGEMN  \
0  31127.0       4.0       2.0       2.0       1.0  5.397605e-79      11.0   
1  31128.0       4.0       2.0       1.0       2.0  1.100000e+01     132.0   
2  31129.0       4.0       2.0       2.0       1.0  1.500000e+01     189.0   
3  31130.0       4.0       2.0       2.0       2.0  8.500000e+01       NaN   
4  31131.0       4.0       2.0       2.0       2.0  4.400000e+01     535.0   
5  31132.0       4.0       2.0       2.0       1.0  7.000000e+01     842.0   
6  31133.0       4.0       2.0       2.0       2.0  1.600000e+01     193.0   
7  31134.0       4.0       2.0       2.0       1.0  7.300000e+01     882.0   
8  31135.0       4.0       2.0       2.0       1.0  5.397605e-79      10.0   
9  31136.0       4.0       1.0       NaN       2.0  4.100000e+01     493.0   

   RIDAGEEX  RIDRETH1  DMQMILIT  ...  FIAPROXY  FIAINTRP  MIALANG  MIAPROXY  \
0      12.0       3.0       NaN  ...       2.0       2.0      NaN       NaN   
1     132.0       4.0       NaN  ...       2.0       2.0      1.0       2.0   
2     190.0       4.0       NaN  ...       2.0       2.0      1.0       2.0   
3       NaN       3.0       2.0  ...       2.0       2.0      NaN       NaN   
4     536.0       4.0       2.0  ...       2.0       2.0      1.0       2.0   
5     843.0       3.0       1.0  ...       2.0       2.0      1.0       2.0   
6     194.0       4.0       NaN  ...       2.0       2.0      1.0       2.0   
7     883.0       3.0       1.0  ...       2.0       2.0      1.0       2.0   
8      11.0       5.0       NaN  ...       2.0       2.0      NaN       NaN   
9       NaN       4.0       2.0  ...       2.0       2.0      NaN       NaN   

   MIAINTRP  AIALANG      WTINT2YR      WTMEC2YR  SDMVPSU  SDMVSTRA  
0       NaN      NaN   6434.950248  6.571396e+03      2.0      44.0  
1       2.0      1.0   9081.700761  8.987042e+03      1.0      52.0  
2       2.0      1.0   5316.895215  5.586719e+03      1.0      51.0  
3       NaN      NaN  29960.839509  3.403099e+04      2.0      46.0  
4       2.0      1.0  26457.708180  2.677058e+04      1.0      48.0  
5       2.0      1.0  32961.509920  3.531554e+04      2.0      52.0  
6       2.0      1.0   5635.221296  5.920618e+03      1.0      51.0  
7       2.0      1.0  43718.506372  4.423117e+04      2.0      48.0  
8       NaN      NaN  15711.544398  1.614382e+04      2.0      52.0  
9       NaN      NaN  26487.611553  5.397605e-79      2.0      51.0  

[10 rows x 43 columns]

Demographics variables:
- DMDBORN4: Due to the concerns of disclosure risk, starting in 2011, country of birth was recoded into two categories: 1) Born in 50 U.S. states or Washington, DC; and 2) Born in other countries, including U.S. territories.

- DMDEDUC2: This variable is the highest grade or level of education completed by adults 20 years and older. The response categories are: less than 9th grade education, 9-11th grade education (includes 12th grade and no diploma), High school graduate/GED, some college or associates (AA) degree, and college graduate or higher.

- DMDHHSIZ: This variable is the number of people in the participant’s household. The values for this variable range from 1 to 7. Due to disclosure concerns, households that are comprised of 7 or more people are included in the category that is labeled ‘7 or more’.

- RIDAGEYR: Age in years, at the time of the screening interview, is reported for survey participants between the ages of 1 and 79 years of age. All responses of participants aged 80 years and older are coded as ‘80.’

- RIAGENDR: Gender: male 1, female 2

- INDHHIN2: Total household income (reported as a range value in dollars), range 1-15, 77 refused, 99 dont know

- DMQMIL1Z: Served active duty in US Armed Forces, 1 yes, 0 no, 7 refuse, 9 dont know

- DMDMARTL: Marital status - 1 Married, 2 widowed, 3 divorced, 4 separated, 5 never married, 6 living with partner, 77 refused	


In [184]:
# filter only relevant cols
DEMO_D = DEMO_2005_2006.filter(['SEQN','RIDRETH1', 'DMDEDUC2', 'DMDHHSIZ', 'RIDAGEYR', 'RIAGENDR','INDHHIN2','INDHHINC', 
                        'DMQMIL1Z', 'DMDMARTL'], axis=1)

In [185]:
# rename col to be more useful
demo_cols = {'SEQN': 'SEQN', 'RIDRETH1':'race_ethnicity', 'DMDEDUC2': 'edu_level', 'DMDHHSIZ': '#_ppl_household', 
               'RIDAGEYR': 'age', 'RIAGENDR': 'gender','INDHHIN2': 'annual_HI', 'INDHHINC':'annual_HI',
                        'DMQMIL1Z': 'served_army', 'DMDMARTL':'marital_status'}

DEMO_D.columns = DEMO_D.columns.map(demo_cols)

In [186]:
DEMO_D.head()

SEQN  race_ethnicity  edu_level  #_ppl_household           age  gender  \
0  31127.0             3.0        NaN              4.0  5.397605e-79     1.0   
1  31128.0             4.0        NaN              7.0  1.100000e+01     2.0   
2  31129.0             4.0        NaN              6.0  1.500000e+01     1.0   
3  31130.0             3.0        4.0              1.0  8.500000e+01     2.0   
4  31131.0             4.0        4.0              4.0  4.400000e+01     2.0   

   annual_HI  marital_status  
0        4.0             NaN  
1        8.0             NaN  
2       10.0             5.0  
3        4.0             2.0  
4       11.0             1.0

## Dietary Data

### Dietary Interview - Total Nutrient Intakes, First Day (DR1TOT_D)

In [187]:
DIET_2005_2006 = pd.read_csv('./Dietary/DR1TOT_D.csv')

Dietary variables:
- DR1TCAFF : caffeine - Caffeine (mg) range 0 to 2448
- DRQSDIET - On special diet? - 1,2,9, missing
- DR1TPROT - Protein (gm) - 0 to 869.49, missing
- DR1TCARB - Carbohydrate (gm) - 8.67 to 1423.87, missing
- DR1TSUGR - Total sugars (gm) - 0.13 to 1115.5, missing
- DR1TFIBE - Dietary fiber (gm) - 0 to 136.6, missing
- DR1TTFAT - Total fat (gm) - 0 to 548.38, missing
- DR1TCHOL - Cholesterol (mg) - 0 to 3515, missing
- DR1TVB6 - Vitamin B6 (mg)
- DR1TFOLA - Total folate (mcg) - 0 to 3172, missing
- DR1TVB12 - Vitamin B12 (mcg) - 0 to 110, missing
- DR1TVD - Vitamin D (D2 + D3) (mcg) - 0 to 86.3, missing
- DR1TMAGN - Magnesium (mg) - 0 2725, missing
- DR1TIRON - Iron (mg) - 0 to 130, missing
- DR1TZINC - Zinc (mg) - 0 to 116, missing
- DR1TSELE - Selenium (mcg) - 0 to 986.5, missing
- DR1_320Z - Total plain water drank yesterday (gm) - 0 13440, missing

In [188]:
# filter only relevant cols
DIET_D = DIET_2005_2006.filter(['SEQN', 'DR1TCAFF', 'DRQSDIET', 'DR1TPROT','DR1TCARB','DR1TSUGR','DR1TFIBE',
                               'DR1TTFAT','DR1TCHOL','DR1TVB6','DR1TFOLA','DR1TVB12','DR1TVD',
                               'DR1TMAGN','DR1TIRON','DR1TZINC','DR1TSELE','DR1_320Z'], axis=1)

diet_cols = {'SEQN': 'SEQN', 'DR1TCAFF':'caffeine','DRQSDIET': 'special_diet', 'DR1TPROT':'protein',
             'DR1TCARB':'carb','DR1TSUGR':'sugar','DR1TFIBE':'fiber',
            'DR1TTFAT':'fat','DR1TCHOL':'diet_chol','DR1TVB6':'VitB6','DR1TFOLA':'folate',
             'DR1TVB12':'VitB12','DR1TVD':'VitD', 'DR1TMAGN':'magnesium',
             'DR1TIRON':'iron','DR1TZINC':'zinc','DR1TSELE':'selenium','DR1_320Z':'water'}

DIET_D.columns = DIET_D.columns.map(diet_cols)

In [189]:
DIET_D.head()

SEQN      caffeine  special_diet  protein    carb   sugar  fiber    fat  \
0  31127.0  5.397605e-79           2.0    50.42  171.45  139.02    4.6  61.28   
1  31128.0  3.300000e+01           2.0    32.51  180.88   88.69   10.0  65.05   
2  31129.0  7.000000e+00           2.0   139.89  684.22  498.82   14.5  94.73   
3  31130.0           NaN           NaN      NaN     NaN     NaN    NaN    NaN   
4  31131.0  1.300000e+01           2.0    27.43  233.73  146.01    6.4  50.86   

   diet_chol  VitB6  folate  VitB12  magnesium   iron   zinc  selenium  \
0      365.0  1.190   154.0    5.71      185.0   5.74   6.34      73.3   
1       70.0  1.088   141.0    1.72      150.0   4.55   5.47      35.8   
2      423.0  2.233   338.0    3.98      328.0  15.93  12.02     206.3   
3        NaN    NaN     NaN     NaN        NaN    NaN    NaN       NaN   
4       48.0  0.385   299.0    1.03      144.0   7.04   3.43      36.0   

          water  
0  5.397605e-79  
1  5.397605e-79  
2  5.000000e+02  
3           NaN  
4  1.925600e+02

In [190]:
len(DIET_D.columns)

17

## Questionnaire Data

### Diabetes

variables
- DIQ010 : doc_diabetes - {Have you/Has SP}} ever been told by a doctor or health professional that {you have/{he/she/SP} has} diabetes or sugar diabetes? 1 yes, 2 no, 3 boarderline
- DIQ160 : doc_prediabetes - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/SP has} any of the following: prediabetes, impaired fasting glucose, impaired glucose tolerance, borderline diabetes or that {your/her/his} blood sugar is higher than normal but not high enough to be called diabetes or sugar diabetes? 1 yes, 2 no

In [191]:
DIQ_2005_2006 = pd.read_csv('./Questionnaire/DIQ_D.csv')

In [192]:
DIQ_2005_2006.columns

Index(['SEQN', 'DIQ010', 'DID040', 'DIQ220', 'DIQ160', 'DIQ170', 'DIQ180',
       'DIQ190A', 'DIQ190B', 'DIQ190C', 'DIQ200A', 'DIQ200B', 'DIQ200C',
       'DIQ050', 'DID060', 'DIQ060U', 'DID070', 'DIQ230', 'DIQ240', 'DID250',
       'DID260', 'DIQ260U', 'DID270', 'DIQ280', 'DIQ290', 'DIQ300S', 'DIQ300D',
       'DID310S', 'DID310D', 'DID320', 'DID330', 'DID340', 'DID350', 'DIQ350U',
       'DIQ360', 'DIQ080'],
      dtype='object')

In [193]:
DIQ_D = DIQ_2005_2006.filter(['SEQN','DIQ010', 'DIQ160'], axis=1)

In [194]:
# rename col to be more useful
DIQ_cols = {'SEQN':'SEQN', 'DIQ010':'doc_diabetes', 'DIQ160': 'doc_prediabetes'}

DIQ_D.columns = DIQ_D.columns.map(DIQ_cols)

In [195]:
DIQ_D.head()

SEQN  doc_diabetes  doc_prediabetes
0  31128.0           2.0              NaN
1  31129.0           2.0              2.0
2  31130.0           2.0              2.0
3  31131.0           2.0              2.0
4  31132.0           1.0              NaN

### Diet Questionnaire

Variables:
- CBQ505 : fast_Dood_12m - In the past 12 months, did you buy food from fast food or pizza places?
- DBD895 : #meals_not_Dome - During the past 7 days, how many meals {did you/did SP} get that were prepared away from home in places such as restaurants, fast food places, food stands, grocery stores, or from vending machines?
    - 0 None, 1 to 21 range, 5555 more than 21 meals per week, 7777 refused, 9999 dont know
- DBD900 : #meals_Dast_Dood - How many of those meals {did you/did SP} get from a fast-food or pizza place?
- DBQ700 : how_Dealthy_diet - In general, how healthy is {your/his/her} overall diet? 
    - Excellent to Poor: 1-5

In [196]:
DBQ_2005_2006 = pd.read_csv('./Questionnaire/DBQ_D.csv')

In [197]:
DBQ_2005_2006.columns

Index(['SEQN', 'DBQ010', 'DBD020', 'DBD030', 'DBD040', 'DBD050', 'DBD060',
       'DBD072A', 'DBD072B', 'DBD072C', 'DBD072D', 'DBD072U', 'DBD080',
       'DBQ700', 'DBQ197', 'DBD222A', 'DBD222B', 'DBD222C', 'DBD222D',
       'DBD222U', 'DBQ229', 'DBQ235A', 'DBQ235B', 'DBQ235C', 'DBQ301',
       'DBQ330', 'DBQ360', 'DBQ370', 'DBD381', 'DBQ390', 'DBQ400', 'DBD411',
       'DBQ421', 'DBQ424', 'DBD091', 'DBQ720', 'DBQ730', 'DBQ740', 'DBQ750',
       'DBQ760', 'DBQ770', 'DBQ780', 'DBQ790', 'DBQ800', 'DBQ810', 'DBQ820',
       'DBQ830', 'DBQ840', 'DBQ850', 'DBQ860', 'DBQ870', 'DBQ880', 'DBQ890'],
      dtype='object')

In [198]:
DBQ_D = DBQ_2005_2006.filter(['SEQN','DBD900', 'DBQ700'], axis=1)

In [199]:
# rename col to be more useful
DBQ_cols = {'SEQN':'SEQN',
                  'DBD900':'#meals_fast_food', 'DBQ700': 'how_healthy_diet'}

DBQ_D.columns = DBQ_D.columns.map(DBQ_cols)

In [200]:
DBQ_D.head()

SEQN  how_healthy_diet
0  31127.0               NaN
1  31128.0               NaN
2  31129.0               NaN
3  31130.0               3.0
4  31131.0               3.0

### Sight & Hearing

variables:
- DLQ010 : difficult_Dearing - 1 yes, 2 no
- DLQ020 : difficult_seeing - 1 yes, 2 no
- DLQ050 - Have serious difficulty walking ? 
    - 1=yes, 2=no, 7/9
- DLQ080 - Have difficulty doing errands alone ? 
    - 1=yes, 2=no, 7/9


In [201]:
# DLQ_2005_2006 = pd.read_csv('./Questionnaire/DLQ_D.csv')

In [202]:
# DLQ_2005_2006.columns

In [203]:
# DLQ_D = DLQ_2005_2006.filter(['SEQN','DLQ010','DLQ020','DLQ050','DLQ080'], axis=1)

In [204]:
# rename col to be more useful
# DLQ_cols = {'SEQN':'SEQN', 'DLQ010': 'difficult_Dearing', 'DLQ020': 'difficult_seeing',
#           'DLQ050':'difficult_walking','DLQ080':'difficult_Drrands'}

# DLQ_D.columns = DLQ_D.columns.map(DLQ_cols)

In [205]:
# DLQ_D.head()

### Drugs

variables: 
- DUQ211 : used_marijuana - Have you ever smoked marijuana or hashish at least once a month for more than one year?
- DUQ230 : # days used marijuana or hashish/month
- DUQ240 : used_CMH - Have you ever used cocaine, crack cocaine, heroin, or methamphetamine?
- DUQ370 : injected_drugs - Have you ever, even once, used a needle to inject a drug not prescribed by a doctor?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [209]:
DUQ_2005_2006 = pd.read_csv('./Questionnaire/DUQ_D.csv')

In [210]:
DUQ_D = DUQ_2005_2006.filter(['SEQN', 'DUQ240'], axis=1)

In [211]:
DUQ_cols = {'SEQN':'SEQN', 'DUQ240':'used_CMH'}

DUQ_D.columns = DUQ_D.columns.map(DUQ_cols)

In [212]:
DUQ_D.head()

SEQN  used_CMH
0  31131.0       2.0
1  31144.0       2.0
2  31151.0       2.0
3  31152.0       2.0
4  31153.0       1.0

### Health Care

HSQ_D
- HSD010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know

HUQ_D
- HUQ010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know
- HUQ041 / 040 - Type place most often go for healthcare
1- Clinic or health center
2 - Doctor's office or HMO
3 - Hospital emergency room
4 - Hospital outpatient department	
5 - Some other place
6 - Doesn't go to one place most often
77 - Refused, 99 - Don't know
- HUQ051/050 - #times receive healthcare over past year
0 - 8 are ranges, 77, 99
- HUQ090 - Seen mental health professional/past yr
1, 2, 7, 9


In [216]:
HUQ_2005_2006 = pd.read_csv('./Questionnaire/HUQ_D.csv')

HUQ_D = HUQ_2005_2006.filter(['SEQN', 'HUQ010', 'HUQ040','HUQ050'], axis=1)
HUQ_cols = {'SEQN':'SEQN', 'HUQ010': 'gen_health_cond','HUQ040':'place_healthcare',
            'HUQ050':'healthcare_past_yr'}

HUQ_D.columns = HUQ_D.columns.map(HUQ_cols)

In [217]:
HUQ_2005_2006.head()

SEQN  HUQ010  HUQ020  HUQ030  HUQ040        HUQ050  HUQ060  HUQ071  \
0  31127.0     1.0     NaN     1.0     2.0  4.000000e+00     NaN     1.0   
1  31128.0     4.0     3.0     1.0     2.0  2.000000e+00     NaN     2.0   
2  31129.0     3.0     3.0     1.0     2.0  5.397605e-79     1.0     2.0   
3  31130.0     3.0     3.0     2.0     NaN  1.000000e+00     NaN     2.0   
4  31131.0     3.0     1.0     1.0     2.0  5.000000e+00     NaN     1.0   

   HUD080  HUQ090  
0     1.0     NaN  
1     NaN     2.0  
2     NaN     2.0  
3     NaN     2.0  
4     2.0     2.0

In [218]:
HUQ_D.head()

SEQN  gen_health_cond  place_healthcare  healthcare_past_yr
0  31127.0              1.0               2.0        4.000000e+00
1  31128.0              4.0               2.0        2.000000e+00
2  31129.0              3.0               2.0        5.397605e-79
3  31130.0              3.0               NaN        1.000000e+00
4  31131.0              3.0               2.0        5.000000e+00

### Health Insurance

variables:
- HIQ011 : health_insurance - {Are you/Is SP} covered by health insurance or some other kind of health care plan? 1 yes, 2 no

In [219]:
HIQ_2005_2006 = pd.read_csv('./Questionnaire/HIQ_D.csv')

HIQ_D = HIQ_2005_2006.filter(['SEQN', 'HIQ011'], axis=1)
HIQ_cols = {'SEQN':'SEQN', 'HIQ011': 'health_insurance'}

HIQ_D.columns = HIQ_D.columns.map(HIQ_cols)

In [220]:
HIQ_D.head()

SEQN  health_insurance
0  31127.0               1.0
1  31128.0               1.0
2  31129.0               1.0
3  31130.0               1.0
4  31131.0               1.0

### Househole Characteristics

HOQ_D
- HOD050 - Number of rooms in home
1-13, 777, 999
- HOQ065 - Home owned, bought, rented, other
1 - Owned or being bought
2 - Rented
3 - Other arrangements
7 - Refused, 9 - Don't know


In [221]:
HOQ_2005_2006 = pd.read_csv('./Questionnaire/HOQ_D.csv')

HOQ_D = HOQ_2005_2006.filter(['SEQN', 'HOD050','HOQ065'], axis=1)
HOQ_cols = {'SEQN':'SEQN', 'HOD050': '#_rooms','HOQ065':'home_ownership'}

HOQ_D.columns = HOQ_D.columns.map(HOQ_cols)

In [222]:
HOQ_D.head()

SEQN  #_rooms  home_ownership
0  31127.0      4.0             2.0
1  31128.0      7.0             1.0
2  31129.0     10.0             1.0
3  31130.0      1.0             2.0
4  31131.0      8.0             1.0

### Early Childhood

ECQ_D
- ECQ020 - Mother smoked when pregnant
1=yes, 2=no, 7/9

In [223]:
ECQ_2005_2006 = pd.read_csv('./Questionnaire/ECQ_D.csv')

ECQ_D = ECQ_2005_2006.filter(['SEQN', 'ECQ020'], axis=1)
ECQ_cols = {'SEQN':'SEQN', 'ECQ020': 'mom_smoked_preg'}

ECQ_D.columns = ECQ_D.columns.map(ECQ_cols)

In [224]:
ECQ_D.head()

SEQN  mom_smoked_preg
0  31127.0              2.0
1  31128.0              1.0
2  31129.0              2.0
3  31135.0              2.0
4  31137.0              2.0

### Food Security

FSQ_D
- FSD032A - HH Worried run out of food
    - 1 - Often true
    - 2 - Sometimes true
    -  - Never true
    - 7 - Refused, 9 - Don't know
- FSD032B - HH Food didn't last
    - “ “
- FSD032C - HH Couldn't afford balanced meals
    - “ “
- FSD041 - HH Adults cut size or skip meals
    - 1, 2, 7, 9
- FSD032D - HH Relied on low-cost food for child
1 - Often true
2 - Sometimes true
3 - Never true
7 - Refused, 9 - Don't know
- FSDHH - Household food security category
1 - HH full food security: 0
2 - HH marginal food security: 1-2
3 - HH low food security: 3-5 (HH w/o child) / 3-7 (HH w/ child)
4 - HH very low food security: 6-10 (HH w/o child) / 8-18 (HH w/ child)
No 7/9, just missing
- FSQ012 / 171 - HH FS benefit: receive in last 12 months
1, 2, 7, 9
- FSQ162 - HH WIC benefit: receive in last 12 month
1,2,7,9


In [225]:
FSQ_2005_2006 = pd.read_csv('./Questionnaire/FSQ_D.csv')

FSQ_D = FSQ_2005_2006.filter(['SEQN', 'FSD032A', 'FSD032B', 'FSD032C', 'FSD041', 'FSD032D', 'FSDHH',
                             'FSQ171','FSD180', 'FSQ162'], axis=1)

 
FSQ_cols = {'SEQN': 'SEQN', 'FSD032A':'run_out_food', 'FSD032B':'food_didnt_last', 'FSD032C':'afford_balanced_meals', 
             'FSD041':'adults_skip_meals', 'FSD032D':'relied_cheap_food', 'FSDHH':'food_security_cat',
             'FSQ171':'food_stamps', 'FSD180':'food_stamps','FSQ162':'WIC'}

FSQ_D.columns = FSQ_D.columns.map(FSQ_cols)

In [226]:
FSQ_D.head()

SEQN  run_out_food  food_didnt_last  afford_balanced_meals  \
0  31127.0           2.0              2.0                    2.0   
1  31128.0           3.0              3.0                    3.0   
2  31129.0           3.0              3.0                    3.0   
3  31130.0           3.0              3.0                    3.0   
4  31131.0           3.0              3.0                    3.0   

   adults_skip_meals  relied_cheap_food  food_security_cat  food_stamps  WIC  
0                2.0                3.0                3.0          2.0  1.0  
1                NaN                3.0                1.0          1.0  2.0  
2                NaN                3.0                1.0          2.0  2.0  
3                NaN                NaN                1.0          2.0  NaN  
4                NaN                3.0                1.0          2.0  2.0

### Income

INQ_D
- IND235 - Monthly family income
    - 1-12 ranges, 77, 99
- INQ244 - Family has savings more than $5000
    - 1,2,7,9


In [227]:
# INQ_2005_2006 = pd.read_csv('./Questionnaire/INQ_D.csv')

# INQ_D = INQ_2005_2006.filter(['SEQN', 'IND235', 'INQ244'], axis=1)

# INQ_cols = {'SEQN': 'SEQN', 'IND235':'monthly_income', 'INQ244':'savings_5000'}

# INQ_D.columns = INQ_D.columns.map(INQ_cols)


In [228]:
# INQ_D.head()

### Medical Conditions

variables:
- MCQ010 : doc_asthma - Has a doctor or other health professional ever told {you/SP} that {you have/s/he/SP has} asthma (az-ma)?
- MCQ080 : doc_overweight - Has a doctor or other health professional ever told {you/SP} that {you were/s/he/SP was} overweight?
- MCQ160a :  doc_arthritis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had arthritis (ar-thry-tis)?
- MCQ160b :  doc_CHF - Has a doctor or other health professional ever told {you/SP} that {you/s/he}had congestive heart failure?
- MCQ160c : doc_CHD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had coronary (kor-o-nare-ee) heart disease?
- MCQ160e : doc_Deart_attack - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had a heart attack (also called myocardial infarction (my-o-car-dee-al in-fark-shun))?
- MCQ160f : doc_stroke - Has a doctor or other health professional ever told {you/SP} that {you/s/her} had a stroke?
- MCQ160k : doc_chronic_bronchitis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had chronic bronchitis?
- MCQ160l : doc_liver_condition - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had any kind of liver condition?
- MCQ160m : doc_thyroid_problem - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had another thyroid (thigh-roid) problem?
- MCQ160o : doc_COPD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had COPD?
- MCQ220 : doc_cancer - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had cancer or a malignancy (ma-lig-nan-see) of any kind?

((- MCQ365a : doc_lose_weight - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: control {your/his/her} weight or lose weight?
- MCQ365b : doc_Dxercise - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: increase {your/his/her} physical activity or exercise?))

- MCQ053 - Taking treatment for anemia/past 3 mos
    - 1,2,7,9
- MCQ084 - Difficulties in thinking or remembering
    - 1, 2, 7, 9
- MCQ230a - What kind of cancer
    - 10-39 different kinds, 66 = more than 3, 77, 99


In [229]:
MCQ_2005_2006 = pd.read_csv('./Questionnaire/MCQ_D.csv')

MCQ_D = MCQ_2005_2006.filter(['SEQN', 'MCQ010', 'MCQ080', 'MCQ160A', 'MCQ160B', 'MCQ160C', 'MCQ160E',
 'MCQ160F', 'MCQ160K', 'MCQ160L', 'MCQ160M', 'MCQ160O', 'MCQ220','MCQ053',
                             'MCQ084','MCQ230A'], axis=1)

 
MCQ_cols = {'SEQN': 'SEQN', 'MCQ010':'doc_asthma', 'MCQ080':'doc_overweight', 'MCQ160A':'doc_arthritis', 
 'MCQ160B':'doc_CHF', 'MCQ160C':'doc_CHD', 'MCQ160E':'doc_heart_attack',
 'MCQ160F':'doc_stroke', 'MCQ160K':'doc_chronic_bronchitis', 'MCQ160L':'doc_liver_condition', 
 'MCQ160M':'doc_thyroid_problem', 'MCQ160O':'doc_COPD', 'MCQ220':'doc_cancer','MCQ053': 'anemia',
                             'MCQ084':'difficult_remembering','MCQ230A':'cancer_type'}

MCQ_D.columns = MCQ_D.columns.map(MCQ_cols)

In [230]:
MCQ_D.head()

SEQN  doc_asthma  doc_overweight  doc_arthritis  doc_CHF  doc_CHD  \
0  31128.0         2.0             NaN            NaN      NaN      NaN   
1  31129.0         2.0             NaN            NaN      NaN      NaN   
2  31130.0         2.0             2.0            2.0      2.0      2.0   
3  31131.0         2.0             2.0            2.0      1.0      2.0   
4  31132.0         2.0             2.0            2.0      2.0      2.0   

   doc_heart_attack  doc_stroke  doc_chronic_bronchitis  doc_liver_condition  \
0               NaN         NaN                     NaN                  NaN   
1               NaN         NaN                     NaN                  NaN   
2               2.0         2.0                     2.0                  2.0   
3               2.0         2.0                     2.0                  2.0   
4               2.0         2.0                     2.0                  2.0   

   doc_thyroid_problem  doc_cancer  anemia  cancer_type  
0                  NaN         NaN     2.0          NaN  
1                  NaN         NaN     2.0          NaN  
2                  2.0         2.0     2.0          NaN  
3                  2.0         2.0     2.0          NaN  
4                  1.0         2.0     2.0          NaN

In [231]:
len(MCQ_D.columns)

14

### Kidney Conditions - Urology (KIQ_U_D)

variables
- KIQ022: doc_kidney - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had weak or failing kidneys? Do not include kidney stones, bladder infections, or incontinence.

In [232]:
KIQ_U_2005_2006 = pd.read_csv('./Questionnaire/KIQ_U_D.csv')

KIQ_U_D = KIQ_U_2005_2006.filter(['SEQN', 'KIQ022'], axis=1)
KIQ_U_cols = {'SEQN':'SEQN', 'KIQ022': 'doc_kidney'}

KIQ_U_D.columns = KIQ_U_D.columns.map(KIQ_U_cols)

In [233]:
KIQ_U_D.head()

SEQN  doc_kidney
0  31130.0         2.0
1  31131.0         2.0
2  31132.0         2.0
3  31134.0         2.0
4  31136.0         2.0

### Occupation

OCQ_D
- OCD150 - Type of work done last week
1 - Working at a job or business
2 - With a job or business but not at work
3 - Looking for work, or
4 - Not working at a job or business?
7 - Refused, 9-  Don’t know
- OCQ180 - Hours worked last week at all jobs
    - Range 0-120 hours
    - 77777, 999999


In [234]:
OCQ_2005_2006 = pd.read_csv('./Questionnaire/OCQ_D.csv')

OCQ_D = OCQ_2005_2006.filter(['SEQN', 'OCD150','OCQ180'], axis=1)
OCQ_cols = {'SEQN':'SEQN', 'OCD150': 'work_type','OCQ180':'hrs_worked_week'}

OCQ_D.columns = OCQ_D.columns.map(OCQ_cols)

In [235]:
OCQ_D.head()

SEQN  work_type  hrs_worked_week
0  31130.0        4.0              NaN
1  31131.0        1.0             45.0
2  31132.0        1.0             15.0
3  31133.0        4.0              NaN
4  31134.0        4.0              NaN

### Oral Health

OHQ_D
- OHQ680 - Last yr embarrassed because of mouth
    - 1 very often to 5 never, 7, 9


In [236]:
OHQ_2005_2006 = pd.read_csv('./Questionnaire/OHQ_D.csv')

OHQ_D = OHQ_2005_2006.filter(['SEQN', 'OHQ680'], axis=1)
OHQ_cols = {'SEQN':'SEQN', 'OHQ680': 'embarassed_mouth'}

OHQ_D.columns = OHQ_D.columns.map(OHQ_cols)

In [237]:
OHQ_D.head()

SEQN  embarassed_mouth
0  31130.0               5.0
1  31131.0               5.0
2  31132.0               5.0
3  31134.0               5.0
4  31136.0               5.0

### Osteoperosis

variables:
- OSQ010A : broken_Dip - Has a doctor ever told {you/SP} that {you/s/he} had broken or fractured {your/his/her} . . .hip?
- OSQ060 : doc_osteoporosis - Has a doctor ever told {you/SP} that {you/s/he} had osteoporosis, sometimes called thin or brittle bones?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [238]:
OSQ_2005_2006 = pd.read_csv('./Questionnaire/OSQ_D.csv')

OSQ_D = OSQ_2005_2006.filter(['SEQN', 'OSQ010A', 'OSQ060'], axis=1)
OSQ_cols = {'SEQN':'SEQN', 'OSQ010A': 'broken_hip', 'OSQ060':'doc_osteoporosis'}

OSQ_D.columns = OSQ_D.columns.map(OSQ_cols)

In [239]:
OSQ_2005_2006.columns

Index(['SEQN', 'OSQ010A', 'OSQ010B', 'OSQ010C', 'OSQ020A', 'OSQ020B',
       'OSQ020C', 'OSD030AA', 'OSQ040AA', 'OSD050AA',
       ...
       'OSQ140U', 'OSQ150', 'OSQ160A', 'OSQ160B', 'OSQ170', 'OSQ180', 'OSQ190',
       'OSQ200', 'OSQ210', 'OSQ220'],
      dtype='object', length=118)

In [240]:
OSQ_D.head()

SEQN  broken_hip  doc_osteoporosis
0  31130.0         2.0               2.0
1  31131.0         2.0               2.0
2  31132.0         2.0               2.0
3  31134.0         2.0               2.0
4  31136.0         2.0               2.0

### Physical Activity

variables:
- PAQ650 - Vigorous recreational activities - In a typical week {do you/does SP} do any vigorous-intensity sports, fitness, or recreational activities that cause large increases in breathing or heart rate like running or basketball for at least 10 minutes continuously? 1 yes 2 no
    - PAD200 - Vigorous activity over past 30 days ^^
- PAQ665 - Moderate recreational activities
    - PAD320 - Moderate activity over past 30 days ^^
- PAD680 : min_sedetary - The following question is about sitting at school, at home, getting to and from places, or with friends including time spent sitting at a desk, traveling in a car or bus, reading, playing cards, watching television, or using a computer. Do not include time spent sleeping. How much time {do you/does SP} usually spend sitting on a typical day?
- PAQ706 - Days physically active at least 60 min.
    - 1-7, 77 99
- PAQ710 - Hours watch TV or videos past 30 days
    - 1-5, 8=don’t watch, 77, 99
    - PAD590
- PAQ715 - Hours use computer past 30 days (on average/day)
    - PAD600
- PAQ722 - Any physical activity in past 7 days

In [241]:
PAQ_2005_2006 = pd.read_csv('./Questionnaire/PAQ_D.csv')

PAQ_D = PAQ_2005_2006.filter(['SEQN', 'PAQ650', 'PAD200', 'PAQ665', 'PAD320',
           'PAD680', 'PAQ706', 'PAQ710','PAD590','PAQ715','PAD600','PAQ722'], axis=1)

PAQ_cols = {'SEQN':'SEQN', 'PAQ650':'vigorous_activity', 'PAD200':'vigorous_activity', 'PAQ665':'moderate_activity',
           'PAD320':'moderate_activity', 'PAD680':'min_sedetary', 'PAQ706':'days_60min_active', 
            'PAQ710':'hours_TV','PAD590':'hours_TV','PAD600':'hours_computer','PAQ722':'activity_week'}

PAQ_D.columns = PAQ_D.columns.map(PAQ_cols)

In [242]:
PAQ_D.columns

Index(['SEQN', 'vigorous_activity', 'moderate_activity', 'hours_TV',
       'hours_computer'],
      dtype='object')

In [243]:
PAQ_D.head() # activity per week missing

SEQN  vigorous_activity  moderate_activity  hours_TV  hours_computer
0  31128.0                NaN                NaN       4.0    5.397605e-79
1  31129.0                1.0                1.0       3.0    2.000000e+00
2  31130.0                2.0                2.0       2.0    6.000000e+00
3  31131.0                2.0                2.0       2.0    5.397605e-79
4  31132.0                2.0                1.0       2.0    5.397605e-79

### Physical Functioning

PFQ_D
- PFQ049 - Limitations keeping you from working
    - 1,2,7,9
- PFQ059 - Physical, mental, emotional limitations
    - 1,2,7,9
- PFQ061A - Managing money difficulty
    - 1 no difficulty to 4 unable to do, 5 don’t do it, 7, 9

Many other activities I could include

In [244]:
PFQ_2005_2006 = pd.read_csv('./Questionnaire/PFQ_D.csv')

PFQ_D = PFQ_2005_2006.filter(['SEQN', 'PFQ049', 'PFQ059', 'PFQ061A'], axis=1)

PFQ_cols = {'SEQN':'SEQN', 'PFQ049':'limit_work', 'PFQ059':'PME_limit', 'PFQ061A':'manage_$_difficult'}

PFQ_D.columns = PFQ_D.columns.map(PFQ_cols)

In [245]:
PFQ_D.head()

SEQN  limit_work  PME_limit  manage_$_difficult
0  31128.0         NaN        NaN                 NaN
1  31129.0         NaN        NaN                 NaN
2  31130.0         2.0        2.0                 1.0
3  31131.0         2.0        2.0                 NaN
4  31132.0         2.0        2.0                 1.0

### Sleeping Disorders

variables:
- SLQ060 : doc_sleeping_disorder - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/s/he has} a sleep disorder?
- SLD010H - How much sleep do you get (hours)?
    - 2-12, 77, 99

In [246]:
SLQ_2005_2006 = pd.read_csv('./Questionnaire/SLQ_D.csv')

SLQ_D = SLQ_2005_2006.filter(['SEQN', 'SLQ060','SLD010H'], axis=1)
SLQ_cols = {'SEQN':'SEQN', 'SLQ060': 'doc_sleeping_disorder','SLD010H':'hrs_sleep'}

SLQ_D.columns = SLQ_D.columns.map(SLQ_cols)

In [247]:
SLQ_2005_2006.columns

Index(['SEQN', 'SLD010H', 'SLD020M', 'SLQ030', 'SLQ040', 'SLQ050', 'SLQ060',
       'SLQ070A', 'SLQ070B', 'SLQ070C', 'SLQ070D', 'SLQ080', 'SLQ090',
       'SLQ100', 'SLQ110', 'SLQ120', 'SLQ130', 'SLQ140', 'SLQ150', 'SLQ160',
       'SLQ170', 'SLQ180', 'SLQ190', 'SLQ200', 'SLQ210', 'SLQ220', 'SLQ230',
       'SLQ240'],
      dtype='object')

In [248]:
SLQ_D.head()

SEQN  doc_sleeping_disorder  hrs_sleep
0  31130.0                    2.0        8.0
1  31131.0                    2.0        9.0
2  31132.0                    2.0        7.0
3  31133.0                    2.0        6.0
4  31134.0                    1.0        7.0

### Smoking - Cigarette Use (SMQ_D)

variables:
- SMQ020 : smoker - These next questions are about cigarette smoking and other tobacco use. {Have you/Has SP} smoked at least 100 cigarettes in {your/his/her} entire life?

In [249]:
SMQ_2005_2006 = pd.read_csv('./Questionnaire/SMQ_D.csv')

SMQ_D = SMQ_2005_2006.filter(['SEQN', 'SMQ020'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ020': 'smoker'}

SMQ_D.columns = SMQ_D.columns.map(SMQ_cols)

In [250]:
SMQ_D.head()

SEQN  smoker
0  31129.0     NaN
1  31130.0     2.0
2  31131.0     2.0
3  31132.0     2.0
4  31133.0     NaN

In [251]:
SMQRTU_2005_2006 = pd.read_csv('./Questionnaire/SMQRTU_D.csv')

# 680 or 681
SMQRTU_D = SMQRTU_2005_2006.filter(['SEQN', 'SMQ680'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ680': 'smoked_5days'}

SMQRTU_D.columns = SMQRTU_D.columns.map(SMQ_cols)

In [252]:
SMQRTU_D.head()

SEQN  smoked_5days
0  31129.0           2.0
1  31130.0           NaN
2  31131.0           2.0
3  31132.0           2.0
4  31133.0           2.0

### Sexual Behavior (SXQ_D)

variables:
- SXQ292 : sexual_orientation_M - Do you think of yourself as...1 Heterosexual or straight (attracted to women), 2 Homosexual or gay (attracted to men), 3 Bisexual (attracted to men and women), 4 Something else, 5 Not sure
- SXQ294 : sexual_orientation_D - ""
- SXQ753 : doc_DPV - Has a doctor or other health care professional ever told you that you had human papillomavirus or HPV? 1 yes, 2 no, 7 refuse, 9 don't know
- SXD171 : lifetime_Demale_partners - In your lifetime, with how many women have you had any kind of sex?
- SXD101 : lifetime_male_partners - In your lifetime, with how many men have you had any kind of sex?


In [253]:
SXQ_2005_2006 = pd.read_csv('./Questionnaire/SXQ_D.csv')

SXQ_D = SXQ_2005_2006.filter(['SEQN', 'SXQ292', 'SXQ294', 'SXD171', 'SXQ410','SXD101', 'SXQ753'], axis=1)
SXQ_cols = {'SEQN':'SEQN', 'SXQ292' : 'sexual_orientation_M', 'SXQ294' : 'sexual_orientation_F', 
           'SXD171' : 'lifetime_female_partners', 'SXD101' : 'lifetime_male_partners', 
            'SXQ410':'lifetime_male_partners',
            'SXQ753' : 'doc_HPV'}

SXQ_D.columns = SXQ_D.columns.map(SXQ_cols)


In [254]:
SXQ_2005_2006.columns

Index(['SEQN', 'SXAISC', 'SXQ021', 'SXD031', 'SXQ101', 'SXQ350', 'SXQ130',
       'SXD171', 'SXQ400', 'SXQ410', 'SXQ430', 'SXQ450', 'SXQ470', 'SXQ490',
       'SXQ510', 'SXQ530', 'SXQ550', 'SXQ570', 'SXQ590', 'SXQ600', 'SXQ610',
       'SXQ251', 'SXQ260', 'SXQ265', 'SXQ270', 'SXQ272', 'SXQ280', 'SXQ292',
       'SXQ294'],
      dtype='object')

In [255]:
SXQ_D.head()

SEQN  sexual_orientation_M  sexual_orientation_F  \
0  31131.0                   NaN                   1.0   
1  31144.0                   1.0                   NaN   
2  31151.0                   NaN                   1.0   
3  31152.0                   NaN                   1.0   
4  31153.0                   NaN                   4.0   

   lifetime_female_partners  lifetime_male_partners  
0                       NaN                     NaN  
1                       6.0            5.397605e-79  
2                       NaN                     NaN  
3                       NaN                     NaN  
4                       NaN                     NaN

notes: may want to concat the # of partners and sexual orientation despite gender

### Taste & Smell

CSQ_D
- CSQ010 - Had problem with smell past 12 months?
    - 1,2,7,9
- CSQ080 - Had problem with taste past 12 months?
    - 1,2,7,9
- CSQ200 - Persistent cold/flu last 12 months?
    - “ “
- CSQ204 - Frequent nasal congestion in past 12 mth

In [256]:
# CSQ_2005_2006 = pd.read_csv('./Questionnaire/CSQ_D.csv')

# CSQ_D = CSQ_2005_2006.filter(['SEQN', 'CSQ010','CSQ080','CSQ200','CSQ204'], axis=1)
# CSQ_cols = {'SEQN':'SEQN', 'CSQ010': 'prob_smell','CSQ080':'prob_tatse',
#             'CSQ200':'persistent_cold','CSQ204':'nasal_congestion'}

# CSQ_D.columns = CSQ_D.columns.map(CSQ_cols)

In [257]:
# CSQ_D.head()

### Alcohol Use (ALQ_D)

Variables: 
- ALQ151 : alcoholic - Was there ever a time or times in {your/SP's} life when {you/he/she} drank {DISPLAY NUMBER} or more drinks of any kind of alcoholic beverage almost every day? 1 yes, 2 no

In [258]:
ALQ_2005_2006 = pd.read_csv('./Questionnaire/ALQ_D.csv')

ALQ_D = ALQ_2005_2006.filter(['SEQN', 'ALQ150'], axis=1)
ALQ_cols = {'SEQN':'SEQN', 'ALQ150': 'alcoholic'}

ALQ_D.columns = ALQ_D.columns.map(ALQ_cols)

In [259]:
ALQ_D.head()

SEQN  alcoholic
0  31130.0        NaN
1  31131.0        2.0
2  31132.0        2.0
3  31134.0        2.0
4  31144.0        2.0

### Weight History

WHQ_D
- WHQ030 - How do you consider your weight
    - 1 - Overweight
    - 2 - Underweight,
    - 3 - About the right weight?
    - 7 - Refused, 9 - Don’t Know
- WHQ070 - Tried to lose weight in past year
    - 1,2,7,9


In [260]:
WHQ_2005_2006 = pd.read_csv('./Questionnaire/WHQ_D.csv')

WHQ_D = WHQ_2005_2006.filter(['SEQN', 'WHQ030','WHQ070'], axis=1)
WHQ_cols = {'SEQN':'SEQN', 'WHQ030':'consider_weight','WHQ070':'tried_lose_weight'}

WHQ_D.columns = WHQ_D.columns.map(WHQ_cols)

In [261]:
WHQ_D.head()

SEQN  consider_weight  tried_lose_weight
0  31130.0              3.0                2.0
1  31131.0              1.0                2.0
2  31132.0              1.0                2.0
3  31133.0              2.0                2.0
4  31134.0              1.0                1.0

## Laboratory Data

### Cholesterol - HDL (HDL_D)

variables:
- LBDHDD : HDL - Direct HDL-Cholesterol (mg/dL), range 10-173

In [262]:
HDL_2005_2006 = pd.read_csv('./Laboratory/HDL_D.csv')

HDL_D = HDL_2005_2006.filter(['SEQN', 'LBDHDD'], axis=1)
HDL_cols = {'SEQN':'SEQN', 'LBDHDD': 'HDL'}

HDL_D.columns = HDL_D.columns.map(HDL_cols)

In [263]:
HDL_D.head()

SEQN   HDL
0  31128.0  55.0
1  31129.0  46.0
2  31130.0   NaN
3  31131.0  39.0
4  31132.0  59.0

### Cholesterol - LDL & Triglycerides (TRIGLY_D)

variables:

- LBDLDL - LBDLDL = ROUND(LBXTC-(LBXHDD+LBXTR/5) for LBXTR less than or equal to 400 mg/dL, and missing for LBXTR greater than 400 mg/dL, range 14-375
- LBXTR : triglyceride - Triglyceride (mg/dL), range 13-4233

In [264]:
TRIGLY_2005_2006 = pd.read_csv('./Laboratory/TRIGLY_D.csv')

TRIGLY_D = TRIGLY_2005_2006.filter(['SEQN', 'LBDLDL', 'LBXTR'], axis=1)
TRIGLY_cols = {'SEQN':'SEQN', 'LBDLDL':'LDL','LBXTR': 'triglyceride'}

TRIGLY_D.columns = TRIGLY_D.columns.map(TRIGLY_cols)

In [265]:
TRIGLY_2005_2006.columns

Index(['SEQN', 'WTSAF2YR', 'LBXTR', 'LBDTRSI', 'LBDLDL', 'LBDLDLSI', 'LBXAPB',
       'LBDAPBSI'],
      dtype='object')

In [266]:
TRIGLY_D.head()

SEQN   LDL  triglyceride
0  31130.0   NaN           NaN
1  31131.0  49.0          86.0
2  31132.0  75.0          65.0
3  31133.0  81.0          61.0
4  31134.0  98.0         195.0

### Cholesterol - Total (TCHOL_D)

variables:
- LBXTC : tot_cholesterol - Total Cholesterol( mg/dL), range 69-813

In [267]:
TCHOL_2005_2006 = pd.read_csv('./Laboratory/TCHOL_D.csv')

TCHOL_D = TCHOL_2005_2006.filter(['SEQN', 'LBXTC'], axis=1)
TCHOL_cols = {'SEQN':'SEQN', 'LBXTC': 'tot_cholesterol'}

TCHOL_D.columns = TCHOL_D.columns.map(TCHOL_cols)

In [268]:
TCHOL_D.head()

SEQN  tot_cholesterol
0  31128.0            129.0
1  31129.0            170.0
2  31130.0              NaN
3  31131.0            105.0
4  31132.0            147.0

### Glycohemoglobin (GHB_D)

variables:
- LBXGH - Glycohemoglobin (%), range 3.5-17.5

In [269]:
GHB_2005_2006 = pd.read_csv('./Laboratory/GHB_D.csv')

GHB_D = GHB_2005_2006.filter(['SEQN', 'LBXGH'], axis=1)
GHB_cols = {'SEQN':'SEQN', 'LBXGH': 'glycohemoglobin'}

GHB_D.columns = GHB_D.columns.map(GHB_cols)

In [270]:
GHB_D.head()

SEQN  glycohemoglobin
0  31129.0              5.2
1  31130.0              NaN
2  31131.0              6.0
3  31132.0              7.1
4  31133.0              4.7

### Herpes Simplex Virus Type-1 & Type-2 (HSV_D)

variables:
- LBXHE2 : herpes_2 - Herpes Simplex Virus Type 2

In [271]:
HSV_2005_2006 = pd.read_csv('./Laboratory/HSV_D.csv')

HSV_D = HSV_2005_2006.filter(['SEQN', 'LBXHE2'], axis=1)
HSV_cols = {'SEQN':'SEQN', 'LBXHE2': 'herpes_2'}

HSV_D.columns = HSV_D.columns.map(HSV_cols)

In [272]:
HSV_D.head(2)

SEQN  herpes_2
0  31129.0       NaN
1  31131.0       1.0

### HIV Antibody Test (HIV_D)

variables:
- LBDHI : HIV - HIV antibody test result, 1 pos, 2, neg, 3 indeterminant

In [273]:
HIV_2005_2006 = pd.read_csv('./Laboratory/HIV_D.csv')

HIV_D = HIV_2005_2006.filter(['SEQN', 'LBDHI'], axis=1)
HIV_cols = {'SEQN':'SEQN', 'LBDHI': 'HIV'}

HIV_D.columns = HIV_D.columns.map(HIV_cols)

In [274]:
HIV_D.head(2)

SEQN  HIV
0  31131.0  2.0
1  31139.0  2.0

### High-Sensitivity C-Reactive Protein (hs-CRP) (HSCRP_I) - NOT IN 2013-2014

variables:
- LBXHSCRP : CRP - HS C-Reactive Protein (mg/L), values 0-188, missing values are mostly below detection limit --> can be made to 0

In [275]:
# HSCRP_2005_2006 = pd.read_csv('./Laboratory/HSCRP_D.csv')

# HSCRP_I = HSCRP_2015_2016.filter(['SEQN', 'LBXHSCRP '], axis=1)
# HSCRP_cols = {'SEQN':'SEQN', 'LBXHSCRP ': 'CRP'}

# HSCRP_I.columns = HSCRP_I.columns.map(HSCRP_cols)

### Serums

CUSEZN_D
- LBXSSE - Serum Selenium (ug/L) - cont range, missing
- LBXSZN - Serum Zinc (ug/dL) - continuous range, missing
- URXUCR - Urinary creatinine (mg/dL) - continuous range, missing

In [276]:
#CUSEZN_2005_2006 = pd.read_csv('./Laboratory/CUSEZN_D.csv')

# CUSEZN_D = CUSEZN_2005_2006.filter(['SEQN', 'LBXSSE','LBXSZN','URXUCR'], axis=1)
# CUSEZN_cols = {'SEQN':'SEQN', 'LBXSSE': 'serum_selenium', 'LBXSZN':'serum_zinc','URXUCR':'serum_creatinine'}

# CUSEZN_D.columns = CUSEZN_D.columns.map(CUSEZN_cols)

In [277]:
# CUSEZN_D.head()

### Folate

FOLATE_D
- LBDRFO - RBC folate (ng/mL) -  continuous range, missing

FOLFMS_D
- LBDFOT - Serum total folate (ng/mL) -  continuous range, missing

In [278]:
FOLATE_2005_2006 = pd.read_csv('./Laboratory/FOLATE_D.csv')
#FOLFMS_2005_2006 = pd.read_csv('./Laboratory/FOLFMS_D.csv')

# LBDRBF or LBDRFO
FOLATE_D = FOLATE_2005_2006.filter(['SEQN', 'LBDRBF','LBXRBF', 'LBDFOL','LBXFOL'], axis=1)
FOLATE_cols = {'SEQN':'SEQN', 'LBDRBF':'rbc_folate','LBXRBF':'rbc_folate','LBDFOL':'serum_folate','LBXFOL':'serum_folate'}
#FOLFMS_D = FOLFMS_2005_2006.filter(['SEQN', 'LBDFOT'], axis=1)
#FOLFMS_cols = {'SEQN':'SEQN', 'LBDFOT':'serum_Dolate'}

FOLATE_D.columns = FOLATE_D.columns.map(FOLATE_cols)
#FOLFMS_D.columns = FOLFMS_D.columns.map(FOLFMS_cols)

In [279]:
# FOLFMS_D.head()

In [280]:
FOLATE_D.head()

SEQN  rbc_folate  serum_folate
0  31128.0       167.0          13.4
1  31129.0       271.0          16.2
2  31130.0         NaN           NaN
3  31131.0       272.0           9.8
4  31132.0       373.0          18.7

### Fasting Glucose

GLU_D
- LBXGLU - Fasting Glucose (mg/dL) - 51 to 421, missing

In [281]:
GLU_2005_2006 = pd.read_csv('./Laboratory/GLU_D.csv')

GLU_D = GLU_2005_2006.filter(['SEQN', 'LBXGLU'], axis=1)
GLU_cols = {'SEQN':'SEQN', 'LBXGLU': 'fasting_glucose'}

GLU_D.columns = GLU_D.columns.map(GLU_cols)

In [282]:
GLU_D.head()

SEQN  fasting_glucose
0  31130.0              NaN
1  31131.0             90.0
2  31132.0            157.0
3  31133.0             84.0
4  31134.0            100.0

### Vitamins B12 and D

VITB12_D
- LBDB12 - Vitamin B12(pg/mL) - range, missing

VID_D
- LBXVIDMS - 25OHD2+25OHD3 (nmol/L) - range, missing (vitamin D)

In [283]:
# VITB12_2005_2006 = pd.read_csv('./Laboratory/VITB12_D.csv')
#  # LBXB12 or LBDB12
# VITB12_D = VITB12_2005_2006.filter(['SEQN', 'LBXB12'], axis=1)
# VITB12_cols = {'SEQN':'SEQN', 'LBXB12': 'lab_VitB12'}

# VITB12_D.columns = VITB12_D.columns.map(VITB12_cols)

VITB12_2005_2006 = pd.read_csv('./Laboratory/B12_D.csv')
VITB12_D = VITB12_2005_2006.filter(['SEQN', 'LBXB12'], axis=1)
VITB12_cols = {'SEQN':'SEQN', 'LBXB12': 'lab_B12'}
VITB12_D.columns = VITB12_D.columns.map(VITB12_cols)


VID_2005_2006 = pd.read_csv('./Laboratory/VID_D.csv')

VID_D = VID_2005_2006.filter(['SEQN', 'LBXVIDMS','LBDVIDMS'], axis=1)
VID_cols = {'SEQN':'SEQN', 'LBXVIDMS': 'lab_VitD','LBDVIDMS':'lab_VitD'}

VID_D.columns = VID_D.columns.map(VID_cols)


In [284]:
VITB12_D.head()

SEQN  lab_B12
0  31128.0   1190.0
1  31129.0    964.0
2  31130.0      NaN
3  31131.0    510.0
4  31132.0    752.0

In [285]:
VID_D.head()

SEQN  lab_VitD
0  31128.0      32.6
1  31129.0      49.5
2  31130.0       NaN
3  31131.0      37.5
4  31132.0      73.8

## Examination Data

### Blood Pressure (BPX_D)

variables:
- BPXSY1 : systolic_BP - Systolic: Blood pres (1st rdg) mm Hg
- BPXDI1 : diastolic_BP - Diastolic: Blood pres (1st rdg) mm Hg
- BPXPLS - 60 sec. pulse (30 sec. pulse * 2) - 0 to 180, missing

In [286]:
BPX_2005_2006 = pd.read_csv('./Examination/BPX_D.csv')

BPX_D = BPX_2005_2006.filter(['SEQN', 'BPXSY1', 'BPXDI1','BPXPLS'], axis=1)
BPX_cols = {'SEQN':'SEQN', 'BPXSY1': 'systolic_BP', 'BPXDI1' : 'diastolic_BP','BPXPLS':'pulse'}

BPX_D.columns = BPX_D.columns.map(BPX_cols)

In [287]:
BPX_D.head(2)

SEQN  systolic_BP  diastolic_BP  pulse
0  31127.0          NaN           NaN    NaN
1  31128.0        100.0          62.0   84.0

### Body Measures (BMX_D)

variables:
- BMXBMI : BMI - Body Mass Index (kg/m**2), range 12.1 to 82.9
- BMXWAIST : waist_C - Waist Circumference (cm), 40.2 to 177.9

In [288]:
BMX_2005_2006 = pd.read_csv('./Examination/BMX_D.csv')

BMX_D = BMX_2005_2006.filter(['SEQN', 'BMXBMI', 'BMXWAIST'], axis=1)
BMX_cols = {'SEQN':'SEQN', 'BMXBMI': 'BMI', 'BMXWAIST' : 'waist_C'}

BMX_D.columns = BMX_D.columns.map(BMX_cols)

In [289]:
BMX_D.head(2)

SEQN    BMI  waist_C
0  31127.0    NaN      NaN
1  31128.0  17.45     62.8

### Dual-Energy X-ray Absorptiometry - Whole Body (DXX_D)

variables:
- DXDTOBMD : bone_mineral_density - Total Bone Mineral Density (g/cm^2), range 0.559 to 1.548

In [290]:
# DXX_2005_2006 = pd.read_csv('./Examination/DXX_D.csv')

# DXX_D = DXX_2005_2006.filter(['SEQN', 'DXDTOBMD'], axis=1)
# DXX_cols = {'SEQN':'SEQN', 'DXDTOBMD': 'bone_mineral_density'}

# DXX_D.columns = DXX_D.columns.map(DXX_cols)

In [291]:
# DXX_D.head(2)

### Oral Health

OHXREF_D
- OHAREC - Overall recommendation for care
    - 1 - See a dentist immediately
    - 2 - See a dentist within the next 2 weeks	
    - 3 - See a dentist at your earliest convenience
    - 4 - Continue your regular routine care
    - Missing


In [292]:
# OHXREF_2005_2006 = pd.read_csv('./Examination/OHXREF_D.csv')

# OHXREF_D = OHXREF_2005_2006.filter(['SEQN', 'OHAREC'], axis=1)
# OHXREF_cols = {'SEQN':'SEQN', 'OHAREC': 'oral_rec'}

# OHXREF_D.columns = OHXREF_D.columns.map(OHXREF_cols)

In [293]:
# OHXREF_D.head()

# Merging dataframes

In [294]:
from functools import reduce

# list of df
# minus DLQ_D, CSQ_D, DXX_D, CUSEZN_D, FOLFMS_D, OHXREF_D, INQ_D,
data_Drames = [DPQ_D, OSQ_D, DEMO_D, DIET_D, DIQ_D, DBQ_D, DUQ_D, HUQ_D, HIQ_D, 
               HOQ_D, ECQ_D, FSQ_D, MCQ_D, KIQ_U_D, OCQ_D, OHQ_D, VITB12_D,
               PAQ_D, PFQ_D, SLQ_D, SMQ_D, SMQRTU_D, SXQ_D, ALQ_D, WHQ_D, 
               HDL_D, TRIGLY_D, TCHOL_D, GHB_D, HSV_D, HIV_D,
               FOLATE_D, GLU_D, VID_D, BPX_D, BMX_D]

df_2005_2006 = reduce(lambda  left,right: pd.merge(left,right,on=['SEQN'],
                                            how='outer'), data_Drames)

In [295]:
df_2005_2006.depressed

0        0.0
1        0.0
2        0.0
3        0.0
4        0.0
        ... 
10343    NaN
10344    NaN
10345    NaN
10346    NaN
10347    NaN
Name: depressed, Length: 10348, dtype: float64

In [296]:
df_2005_2006.dropna(subset=['depressed'], inplace=True)

In [297]:
df_2005_2006.shape

(4836, 98)

In [298]:
df_2005_2006.columns

Index(['SEQN', 'depressed', 'broken_hip', 'doc_osteoporosis', 'race_ethnicity',
       'edu_level', '#_ppl_household', 'age', 'gender', 'annual_HI',
       'marital_status', 'caffeine', 'special_diet', 'protein', 'carb',
       'sugar', 'fiber', 'fat', 'diet_chol', 'VitB6', 'folate', 'VitB12',
       'magnesium', 'iron', 'zinc', 'selenium', 'water', 'doc_diabetes',
       'doc_prediabetes', 'how_healthy_diet', 'used_CMH', 'gen_health_cond',
       'place_healthcare', 'healthcare_past_yr', 'health_insurance', '#_rooms',
       'home_ownership', 'mom_smoked_preg', 'run_out_food', 'food_didnt_last',
       'afford_balanced_meals', 'adults_skip_meals', 'relied_cheap_food',
       'food_security_cat', 'food_stamps', 'WIC', 'doc_asthma',
       'doc_overweight', 'doc_arthritis', 'doc_CHF', 'doc_CHD',
       'doc_heart_attack', 'doc_stroke', 'doc_chronic_bronchitis',
       'doc_liver_condition', 'doc_thyroid_problem', 'doc_cancer', 'anemia',
       'cancer_type', 'doc_kidney', 'work_type', '

# Export 2005-2006 data

In [299]:
df_2005_2006.to_csv('/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/df_2005_2006_Dxt.csv', index=False)